packages/engine/src/avatar/state/AvatarNetworkState.tsx

```diff
@@ -96,24 +122,40 @@ const AvatarReactor = React.memo(({ entityUUID }: { entityUUID: EntityUUID }) =>
   useEffect(() => {
     if (!state.avatarID.value) return
 
+    let aborted = false
+
     Engine.instance.api
       .service(avatarPath)
       .get(state.avatarID.value)
       .then((avatarDetails) => {
+        if (aborted) return
```

Comment:
We should be passing down a AbortSignal all the way to the fetch() request, so that network requests can be cancelled when they are no longer necessary

Reply:
yes, can do in a follow up when we have better tooling